torch/csrc/DataLoader.cpp

```diff
@@ -59,10 +59,32 @@ SIGNAL_HANDLER(SIGBUS, handler_SIGBUS, "ERROR: Unexpected bus error encountered
   "This might be caused by insufficient shared memory (shm).\n");
 SIGNAL_HANDLER(SIGSEGV, handler_SIGSEGV, "ERROR: Unexpected segmentation fault encountered in worker.\n");
 
+// When an error happend in DataLoader methods and Python starts to  exit, the
+// error trace will keep the loader alive, and Python may kill the children
+// processes first before deleting the loader object. Then the cleaning up
+// methods in DataLoader.__del__ are not yet called, and SIGCHILD will print an
+// error saying a worker is killed by SIGTERM. So we suppress SIGTERM from main
+// loader process here to avoid this.
+static void handler_SIGTERM(int sig, siginfo_t *info, void *ctx)
+{
+  if (info->si_pid == getppid()) {
+    _exit(EXIT_SUCCESS);
```

Comment:
EXIT_SUCCESS seems like a weird code for a process that just got SIGTERMed in the middle of work

Reply:
But it only EXIT_SUCCESS when it got SIGTERM from the parent loader process. Since we don't expose these worker processes to users, this should really only happen when loader process is terminated. If we don't EXIT_SUCCESS, then the non-zero exit code may trigger `RuntimeError` in loader SIGCHLD handler.